packages/dashboard/src/components/sidePanel/utils/index.ts

```diff
@@ -1,18 +1,34 @@
-import { useEffect, Dispatch, useState } from 'react';
+import { Dispatch, useEffect, useState } from 'react';
 // import { useDispatch } from 'react-redux';
-import { get, set, cloneDeep } from 'lodash';
+import { get } from 'lodash';
+import { useSelector } from 'react-redux';
+import { DashboardState } from '../../../store/state';
 
 export const useInput: <T>(path: string) => [T, Dispatch<T>] = (path) => {
-  const selectedWidget = {}; // TODO: get selected widget from state manager or context
-  const [value, updateValue] = useState(get(selectedWidget, path));
-  // const dispatch = useDispatch();
+  const [selectedWidget] = useSelector((state: DashboardState) => state.dashboardConfiguration.widgets);
+  const [value, updateState] = useState(get(selectedWidget, path));
+
+  // update form value when widget status changes. E.g. a user updates a widget `x` by dragging
+  useEffect(() => {
+    const currentWidgetState = get(selectedWidget, path);
+    if (value !== currentWidgetState) {
+      updateState(currentWidgetState);
+    }
+  }, [selectedWidget]);
```

Comment:
You're introducing technical debt here by limiting the observation to the selected widget, instead of all widgets. Could you add a TECHDEBT: comment here?